packages/jest-cli/src/cli/runCLI.js

```diff
@@ -28,57 +28,74 @@ const watch = require('../watch');
 
 const VERSION = require('../../package.json').version;
 
-module.exports = (
+module.exports = async (
   argv: Object,
-  root: Path,
+  roots: Array<Path>,
   onComplete: (results: ?AggregatedResult) => void,
 ) => {
   const realFs = require('fs');
   const fs = require('graceful-fs');
   fs.gracefulify(realFs);
 
   const pipe = argv.json ? process.stderr : process.stdout;
-  argv = argv || {};
   if (argv.version) {
     pipe.write(`v${VERSION}\n`);
     onComplete && onComplete();
     return;
   }
 
-  const _run = async ({config, hasDeprecationWarnings}) => {
+  const _run = async (
+    configs: Array<{config: Config, hasDeprecationWarnings: boolean}>,
+  ) => {
     if (argv.debug) {
-      logDebugMessages(config, pipe);
+      // TODO fix/remove this: there should be a `--show-config` argument.
```

Comment:
We also could make a checklist under https://github.com/facebook/jest/issues/2970 and treat it as an umbrella task